Use .json for JMH-results and rework benchmark_evaluation.py

leicmi opened the pull request:
This PR will set the JMH output format to JSON, which makes it easier to parse the output files and allows the usage of other tools such as [gradle-jmh-report](https://github.com/jzillmann/gradle-jmh-report).

Support for JSON is added `benchmark_evaluation.py`, which has been completely reworked and also supports some command line arguments, e.g. for adding the confidence interval to the plots and disabling the git integration.

Notes:
- If Gradle is run from an IDE like IntelliJ, the python virtual environment might not be used. Either run Gradle from within the venv using the command line or install the dependencies from `requirements.txt` globally.

oertl commented on requirements.txt:
```diff
@@ -0,0 +1,4 @@
+black==22.6.0
```
latest version is 22.8.0

leicmi commented:
Thanks for the review! I rebased and added 3 new commits that should fix the problems with spotless/black:
- [Update black in requirements.txt from 22.6.0 to 22.8.0](https://github.com/dynatrace-oss/hash4j/pull/4/commits/d4a29193fbad8858b6b99a0bf0daa7c229dfc8c9)


- [Reformat using the new black](https://github.com/dynatrace-oss/hash4j/pull/4/commits/9ba4f08e801d3c3ef3f9c147b731b21c84ca3874)


- [Use black from requirements.txt in the GitHub workflow](https://github.com/dynatrace-oss/hash4j/pull/4/commits/aa9febe62a31522814057736e365d43635854b99)
